src/Kmd.Logic.Gateway.Automation/PreValidation/ApiPreValidation.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Gateway;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    public class ApiPreValidation : PreValidationBase, IValidation
+    {
+        public ApiPreValidation(string folderPath)
+            : base(folderPath)
+        {
+        }
+
+        public async Task<bool> ValidateAsync(GatewayDetails gatewayDetails)
+        {
+            var isValidationSuccess = true;
+            if (gatewayDetails != null)
+            {
+                foreach (var api in gatewayDetails.Apis)
+                {
+                    if (string.IsNullOrEmpty(api.Name))
+                    {
+                        this.ValidationResults.Add(new PublishResult { IsError = true, ResultCode = ResultCode.InvalidInput, Message = $"Api Name not exist" });
+                        isValidationSuccess = false;
+                    }
+
+                    if (string.IsNullOrEmpty(api.Path) || !Uri.IsWellFormedUriString(api.Path, UriKind.Absolute))
```

Comment:
Is `!Uri.IsWellFormedUriString(api.Path, UriKind.Absolute)` going to work? `api.Path` is for example: _"studica-demo/programmes"_. It's just a part of the `Uri`.

Reply:
this should be complete backend url, not relative

Reply:
So, not path, but some interpolated string from the model?